### PR TITLE
machine/usb/hid/joystick: move joystick under HID and remove duplicate code

### PR DESCRIPTION
**Pull request by deadprogram**
machine/usb/hid/joystick: move joystick under HID as it belongs and also remove duplicate code.

As discussed here: https://github.com/tinygo-org/tinygo/issues/3474#issuecomment-1478487194

### CODE REVIEW

**Comment by deadprogram** [comment]
@sago35 @nobonobo any thoughts on this?

File: src/machine/usb/hid/hid.go [hunk 1]
```diff
@@ -50,6 +50,8 @@ func handler() {
 	}
 }
 
+var DefaultSetupHandler = setupHandler
```
Comment (sago35) [inline src/machine/usb/hid/hid.go:53]: It cannot be set by overriding DefaultSetHandler.
Perhaps hid.SetHandler needs to be changed as well.

File: src/machine/usb/joystick/joystick.go [hunk 2]
```diff
@@ -1,97 +0,0 @@
...
-		State: def.NewState(),
-	}
-	m.State = def.NewState()
-	if setupFunc == nil {
-		setupFunc = m.setupFunc
-	}
-	machine.EnableJoystick(m.handler, rxHandlerFunc, setupFunc, hidDesc)
-	js = m
-	return m
-}
```
Comment (sago35) [inline src/machine/usb/joystick/joystick.go:36]: This is a necessary part of ForceFeedbackJoystick.
I'll consider a few ideas on how to change it.

Comment (sago35) [reply src/machine/usb/joystick/joystick.go:36]: I would like you to MERGE the following

https://github.com/tinygo-org/tinygo/commit/f3efea9e7a65f50ae29c31e22a94ce6a0bb87351


This allows the ForceFeedbackJoystick to be configured.
The following is the code that will be changed by this PR.

https://github.com/sago35/pico-ffb-steering-wheel/commit/e737234a42ace757eae7efaa85f245ffbf86b54d

Comment (deadprogram) [reply src/machine/usb/joystick/joystick.go:36]: I had a feeling you were using this somewhere. See my suggestion on small rename to your proposed function.

Comment (sago35) [reply src/machine/usb/joystick/joystick.go:36]: Updated

https://github.com/tinygo-org/tinygo/tree/usb-hid-joystick-refactor-sago35

File: src/machine/usb/hid/hid.go (hunk 1 above)
Comment (sago35) [reply src/machine/usb/hid/hid.go:53]: This would probably be a feature addition. So I did not think it was necessary to address it in this PR.

**Review by sago35 — APPROVED** [review]
LGTM

**Comment by deadprogram** [comment]
@sago35 thanks for the review and for adding the fix to this PR as well! Now merging.

**Pull request merged**